Tensile/Source/lib/include/Tensile/ContractionProblemPredicates.hpp

```diff
@@ -1055,6 +1055,29 @@ namespace Tensile
                     return problem.stridedBatched() == value;
                 }
             };
+
+            struct PerCU : public Predicate_CRTP<PerCU, ContractionProblem>
+            {
+                enum
+                {
+                    HasIndex = false,
+                    HasValue = false
+                };
+                bool value;
+
+                PerCU() = default;
+
+                static std::string Type()
+                {
+                    return "PerCU";
+                }
+
+                virtual bool operator()(ContractionProblem const& problem) const override
+                {
+                    return true; // TODO - logic for determining to use per CU or not
```

Comment:
Ultimately some kind of user-accessible switch in the ContractionProblem should be involved here. Something like `deterministicMode` or `arithmeticUnit`.  I think naming it something along the lines of selecting the most efficient solution vs. selecting the fastest solution will be more intuitive.  I wonder if naming this feature in that way ('efficient' or 'efficiency') would be better than 'per-cu'?

Reply:
What about 'CUEfficiency' for the predicate name? Still conveys the idea of selecting fastest vs most efficient, but retains the fact that this is calculated relative to CUs.

Reply:
How about "Global efficiency" versus "Local efficiency" ? Kind of alludes to 'overall' versus 'per-cu'.